lib/CleantalkSP/SpbctWP/Scanner/Cure.php

```diff
@@ -88,8 +88,10 @@ public function signature_cure($file, $signature) {
                     } else {
                         global $spbc;
 
-                        $spbc->settings['there_was_signature_treatment'] = 1;
-                        $spbc->save('settings');
+                        if(!$spbc->settings['there_was_signature_treatment']) {
+                            $spbc->settings['there_was_signature_treatment'] = 1;
```

Comment:
You need to add this _there_was_signature_treatment_ to State::$settings and in Updater\UpdaterScripts::updateTo_2_82_0()

Reply:
Done